packages/editor/src/browser/editor.ts

```diff
@@ -60,6 +62,8 @@ export interface TextEditor extends Disposable, TextEditorSelection, Navigatable
      * @returns identifiers of applied decorations, which can be removed in next call.
      */
     deltaDecorations(params: DeltaDecorationParams): string[];
+
+    getModel(): editor.IModel;
```

Comment:
We don't need a new abstraction as  `Model` in the editor extension, there is already an editor `TextEditor` and its model `TextEditorDocumen`, new APIs should be added on them.